### ISSUE TITLE
Types referenced in attribute arguments are not detected for attributes on enum values

### ISSUE THREAD

**Issue by andersforsgren** [realvizu/NsDepCop#3]
When a type is referenced from the arguments of an attribute, and this attribute is placed on an enum field, violations are not detected.

In the example below the attribute type is defined in the local namespace, but the types `ForbiddenType` and `ForbiddenAttribute` are defined in some namespace that is disallowed in the NsDepCop config.
The use of the forbidden type is detected correctly except in the fifth example, where it is not detected. It appears to be missed only when it is placed on a field (value) of an enum. 

``` csharp
   [AttributeUsage(AttributeTargets.All)]
   class AllowedAttributeWithTypeArg : Attribute
   {
      public AllowedAttributeWithTypeArg(Type t) { }
   }

   class Foo1
   {
      // Direct reference to forbidden type in expression. Detected ok.
      public static Type TypeField = typeof(ForbiddenType);
   }

   [ForbiddenAttribute("Attributes defined in the foreign namespace detected ok")]
   class Foo2
   {   
   }

   // Reference to forbidden type in arg to attribute on class. Detected OK.
   [AllowedAttributeWithTypeArg(typeof(ForbiddenType))]
   class Foo3
   {
   }

   class Foo4
   {
      // Reference to forbidden type in arg to attribute on class field. Detected OK.
      [AllowedAttributeWithTypeArg(typeof(ForbiddenType))]
      public int x;
   }

   enum Foo5
   {
      Value1,
      // Reference to forbidden type in arg to attribute on enum value. Not Detected!
      [AllowedAttributeWithTypeArg(typeof(ForbiddenType))]
      Value2
   }
```

**Comment by realvizu** [comment]
Thanks for catching and reporting this!
I'll try to fix it in the next release.

**Comment by realvizu** [comment]
Fixed. 
It turns out that EnumDeclarationSyntax is not a descendant of TypeDeclarationSyntax but of BaseTypeDeclarationSyntax.
